volatility3/framework/plugins/windows/vadyarascan.py

```diff
@@ -73,26 +73,43 @@ def _generator(self):
                     )
                     continue
 
-                for match in rules.match(data=layer.read(start, size, True)):
-                    if yarascan.YaraScan.yara_returns_instances():
-                        for match_string in match.strings:
-                            for instance in match_string.instances:
+                if not yarascan.YaraScan._yara_x:
+                    for match in rules.match(data=layer.read(start, size, True)):
+                        if yarascan.YaraScan.yara_returns_instances():
+                            for match_string in match.strings:
+                                for instance in match_string.instances:
+                                    yield 0, (
+                                        format_hints.Hex(instance.offset + start),
+                                        task.UniqueProcessId,
+                                        match.rule,
+                                        match_string.identifier,
+                                        instance.matched_data,
+                                    )
+                        else:
+                            for offset, name, value in match.strings:
                                 yield 0, (
-                                    format_hints.Hex(instance.offset + start),
+                                    format_hints.Hex(offset + start),
                                     task.UniqueProcessId,
                                     match.rule,
+                                    name,
+                                    value,
+                                )
+                else:
+                    data = layer.read(start, size, True)
```

Comment:
This is shared between both implementations, as are many of the return values.  It would be nice if we could shrink the differences to just the loop (but it's not critical so if can't be done easily, don't worry about it).